webapp/channels/src/components/post/post_component.tsx

```diff
@@ -513,7 +513,6 @@ const PostComponent = (props: Props): JSX.Element => {
             {(isSearchResultItem || (props.location !== Locations.CENTER && (props.isPinnedPosts || props.isFlaggedPosts))) && <DateSeparator date={currentPostDay}/>}
             <PostAriaLabelDiv
                 ref={postRef}
-                role='listitem'
```

Comment:
This and the changes at [post_list_virtualized/post_list_virtualized.tsx](https://github.com/mattermost/mattermost/pull/24122/files#diff-61c4a17151b88abdf551ba2bc3a2298363473a02fd9352a8b878002fcd831069), and on [dynamic-virtualized-list](https://github.com/mattermost/dynamic-virtualized-list/pull/28) fixed the issue in the screenshot. It properly set the ARIA role's parent-children hierarchy in the post list content.
<img width="884" alt="Screenshot 2023-07-26 at 4 52 14 PM" src="https://github.com/mattermost/mattermost/assets/5334504/4ae993a2-a527-493e-a041-c7c6ae88aa08">